src/client_side.cc

```diff
@@ -3478,6 +3482,14 @@ clientListenerConnectionOpened(AnyP::PortCfgPointer &s, const Ipc::FdNoteId port
            << s->listenConn);
 
     Must(AddOpenedHttpSocket(s->listenConn)); // otherwise, we have received a fd we did not ask for
+
+#if USE_SYSTEMD
+    // at least one port is open, tell systemd we are able to serve connections
+    // subsequent sd_notify() calls will not be processed because first parameter is non-zero
+    // XXX: send the notification only when all configured ports are open
+    if (opt_foreground || opt_no_daemon)
```

Comment:
This code path is shared by startup and reconfigure. A local static counter would avoid sending a burst on reconfigure as well.
```suggestion
    static int portsSeen = 0;
    if (++portsSeen == NHttpSockets && (opt_foreground || opt_no_daemon))
```

NP: this does not resolve the XXX because NHttpSockets only points at the number of sockets that started the opening process. The two counters may match on every port, or at a subset of points during startup. This will be resolved when HttpPortList becomes a proper STL container we can check length() on.